GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5876,17 +5977,113 @@ WindowMsgHandledType IdleWorkerSystem( GameWindow *window, UnsignedInt msg,
 
 }
 
-void InGameUI::drawSystemTime()
+namespace
+{
+	// helpers for inline counters
+	constexpr const Int kHudAnchorX = 3;
+	constexpr const Int kHudAnchorY = -1;
+	constexpr const Int kHudGapPx = 6;
+	inline Bool isAtHudAnchorPos(const Coord2D &p) { return p.x == kHudAnchorX && p.y == kHudAnchorY; }
+	inline Bool isAutoOrAtHudAnchorPos(const Coord2D &p) { return (p.x == -1 && p.y == -1) || isAtHudAnchorPos(p); }
+}
+
+void InGameUI::drawNetworkLatency(Int &x, Int &y)
+{
+	const UnsignedInt networkLatencyFrames = TheNetwork->getRunAhead();
+
+	if (networkLatencyFrames != m_lastNetworkLatencyFrames)
+	{
+		UnicodeString latencyStr;
+		latencyStr.format(L"%u", networkLatencyFrames);
+		m_networkLatencyString->setText(latencyStr);
+		m_lastNetworkLatencyFrames = networkLatencyFrames;
+	}
+
+	// TheSuperHackers @info at the HUD anchor this draws inline and advances x otherwise uses configured position
+	if (isAutoOrAtHudAnchorPos(m_networkLatencyPosition))
+	{
+		m_networkLatencyString->draw(kHudAnchorX + x, kHudAnchorY + y, m_networkLatencyColor, m_networkLatencyDropColor);
+		x += m_networkLatencyString->getWidth() + kHudGapPx;
+	}
+	else
+	{
+		m_networkLatencyString->draw(m_networkLatencyPosition.x, m_networkLatencyPosition.y, m_networkLatencyColor, m_networkLatencyDropColor);
+	}
+}
+
+void InGameUI::drawRenderFps(Int &x, Int &y)
+{
+	const UnsignedInt nowMs = timeGetTime();
```

Comment:
This now again calls `timeGetTime` when m_renderFpsRefreshMs was 0. This should not be necessary I think.